libraries/standard/json/src/json.c

```diff
@@ -0,0 +1,1014 @@
+        }
+
+        ret = true;
+
+        if( skipSpaceAndComma( buf, &i, max ) != true )
+        {
+            break;
+        }
+    }
+
+    *start = i;
+    return ret;
+}
+
+/**
+ * @brief Advance buffer index beyond the scalar key-value pairs
+ * of an object.
+ *
+ * In JSON, objects consist of comma-separated key-value pairs.
+ * A key is always a string (a scalar) while a value may be a
+ * scalar, an object, or an array.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ *
+ * @return true if a key was present;
+ * false otherwise.
+ *
+ * @note Stops advance if a value is an object or array.
+ */
+static bool skipObjectScalars( const char * buf,
+                               size_t * start,
+                               size_t max )
+{
+    bool ret = false;
+    size_t i = *start;
+
+    while( i < max )
+    {
+        if( skipString( buf, &i, max ) != true )
+        {
+            break;
+        }
+
+        skipSpace( buf, &i, max );
+
+        if( ( i < max ) && ( buf[ i ] != ':' ) )
+        {
+            break;
+        }
+
+        i++;
+        ret = true;
+        skipSpace( buf, &i, max );
+
+        if( skipAnyScalar( buf, &i, max ) != true )
+        {
+            break;
+        }
+
+        if( skipSpaceAndComma( buf, &i, max ) != true )
+        {
+            break;
+        }
+    }
+
+    *start = i;
+    return ret;
+}
+
+/**
+ * @brief Advance buffer index beyond one or more scalars.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ *
+ * @return true if a scalar was present;
+ * false otherwise.
+ */
+static bool skipScalars( const char * buf,
+                         size_t * start,
+                         size_t max,
+                         char mode )
+{
+    bool ret = false;
+
+    assert( ( mode == '[' ) || ( mode == '{' ) );
+
+    skipSpace( buf, start, max );
+
+    if( mode == '[' )
+    {
+        ret = skipArrayScalars( buf, start, max );
+    }
+    else
+    {
+        ret = skipObjectScalars( buf, start, max );
+    }
+
+    return ret;
+}
+
+/**
+ * @brief Advance buffer index beyond a collection and handle nesting.
+ *
+ * A stack is used to continue parsing the prior collection type
+ * when a nested collection is finished.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ *
+ * @return #JSONSuccess if the buffer contents are a valid JSON collection;
+ * #JSONIllegalDocument if the buffer contents are NOT valid JSON;
+ * #JSONMaxDepthExceeded if object and array nesting exceeds a threshold;
+ * #JSONPartial if the buffer contents are potentially valid but incomplete.
+ */
+#ifndef JSON_MAX_DEPTH
+    #define JSON_MAX_DEPTH    32
+#endif
+static JSONStatus_t match( const char * buf,
```

Comment:
i would have guessed this would be called skipCollection, which seems more indicative of what it does than "match". Is there a reason for the name difference?